src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -61,10 +83,20 @@ function useTransactionsWithCounterparties () {
   const allCounterparties = uniqBy('id', holofuelInboxCounterparties.concat([whoami]))
 
   const updatedActionableTransactions = updateCounterparties(holofuelActionableTransactions, allCounterparties)
-  const updatedNonPendingTransactions = updateCounterparties(holofuelNonPendingTransactions, allCounterparties)
+  console.log('updatedActionableTransactions', updatedActionableTransactions)
+
+  const updatedActionableWOCanceledOffers = updatedActionableTransactions.filter(actionableTx => (actionableTx.status !== STATUS.canceled) && (actionableTx.type !== TYPE.request))
+  console.log('updatedActionableWOCanceledOffers', updatedActionableWOCanceledOffers)
+
+  const updatedCanceledTransactions = updatedActionableTransactions.filter(actionableTx => actionableTx.status === STATUS.canceled)
+  const updatedDeclinedTransactions = updatedActionableTransactions.filter(actionableTx => actionableTx.status === STATUS.declined)
+  const updatedNonPendingTransactions = updateCounterparties(holofuelNonPendingTransactions, allCounterparties).concat(updatedCanceledTransactions).concat(updatedDeclinedTransactions)
+
+  // console.log('updatedActionableWithoutCanceled : ', updatedActionableWOCanceledOffers)
+  // console.log('updatedNonPendingTransactions : ', updatedNonPendingTransactions)
 
   return {
-    actionableTransactions: updatedActionableTransactions,
+    actionableTransactions: updatedActionableWOCanceledOffers,
```

Comment:
Totally agree about the desire to abstract away the business logic into the dna interface. 

*To document what I shared in our last conversation,  I chose to go with this approach to avoid more endpoint calls that are subject to the interim fetch:network-only method we're using currently to overcoming any polling/caching issues.  Essentially, too many calls, too frequently are occurring with each getall call at the moment.  Let's look at moving this as an endpoint on the interface once we refactor with subscriptions or different caching approaches.*